crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs

```diff
@@ -56,81 +128,317 @@ impl TryFrom<&DeutschebankRouterData<&PaymentsAuthorizeRouterData>>
+                            reference,
+                            signed_on,
+                        }
+                    },
+                })
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum DeutschebankTXAction {
+    Authorization,
+    Capture,
+    Credit,
+    Preauthorization,
+    Refund,
+    Reversal,
+    RiskCheck,
+    #[serde(rename = "verify-mop")]
+    VerifyMop,
+    Payment,
+    AccountInformation,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct BankAccount {
+    account_holder: Option<Secret<String>>,
+    bank_name: Option<Secret<String>>,
+    bic: Option<Secret<String>>,
+    iban: Option<Secret<String>>,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct TransactionBankAccountInfo {
+    bank_account: Option<BankAccount>,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct DeutschebankTransactionInfo {
+    back_state: Option<String>,
+    ip_address: Option<Secret<String>>,
+    #[serde(rename = "type")]
+    pm_type: Option<String>,
+    transaction_bankaccount_info: Option<TransactionBankAccountInfo>,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
 pub struct DeutschebankPaymentsResponse {
-    status: DeutschebankPaymentStatus,
-    id: String,
+    rc: String,
+    message: String,
+    timestamp: String,
+    back_ext_id: Option<String>,
+    back_rc: Option<String>,
+    event_id: Option<String>,
+    kind: Option<String>,
+    tx_action: Option<DeutschebankTXAction>,
+    tx_id: Option<String>,
+    amount_total: Option<DeutschebankAmount>,
+    transaction_info: Option<DeutschebankTransactionInfo>,
 }
 
-impl<F, T> TryFrom<ResponseRouterData<F, DeutschebankPaymentsResponse, T, PaymentsResponseData>>
-    for RouterData<F, T, PaymentsResponseData>
+impl
+    TryFrom<
+        ResponseRouterData<
+            CompleteAuthorize,
+            DeutschebankPaymentsResponse,
+            CompleteAuthorizeData,
+            PaymentsResponseData,
+        >,
+    > for RouterData<CompleteAuthorize, CompleteAuthorizeData, PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: ResponseRouterData<F, DeutschebankPaymentsResponse, T, PaymentsResponseData>,
+        item: ResponseRouterData<
+            CompleteAuthorize,
+            DeutschebankPaymentsResponse,
+            CompleteAuthorizeData,
+            PaymentsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
+        let resource_id = ResponseId::ConnectorTransactionId(
+            item.response
+                .tx_id
+                .ok_or(errors::ConnectorError::MissingConnectorTransactionID)?,
```

Comment:
If connector_transaction_id is missing shall we fail the payment or mark it pending?

Reply:
We are not marking tx_id as option now. Deserialization will happen if tx_id is not in response.